src/gt4py/cartesian/gtc/debug/debug_codegen.py

```diff
@@ -0,0 +1,354 @@
+        data_dimensions: list[str] = [str(dim) for dim in temporary_declaration.data_dims]
+        shape = shape + data_dimensions
+        shape_decl = ", ".join(shape)
+        dtype: str = self.visit(temporary_declaration.dtype)
+        field_offset = tuple(-ext[0] for ext in local_field_extent)
+        offset = [str(off) for off in field_offset] + ["0"] * (
+            1 + len(temporary_declaration.data_dims)
+        )
+        return f"{temporary_declaration.name} = Field.empty(({shape_decl}), {dtype}, ({', '.join(offset)}))"
+
+    def visit_DataType(self, data_type: DataType, **_) -> str:
+        if data_type not in {DataType.BOOL}:
+            return f"np.{data_type.name.lower()}"
+        else:
+            return data_type.name.lower()
+
+    def visit_VariableKOffset(self, variable_k_offset: VariableKOffset, **_) -> str:
+        return f"i,j,k+int({self.visit(variable_k_offset.k)})"
+
+    def visit_CartesianOffset(self, cartesian_offset: CartesianOffset, **kwargs) -> str:
+        dimensions = kwargs["dimensions"]
+        return cartesian_offset.to_str(dimensions)
+
+    def visit_SymbolRef(self, symbol_ref: SymbolRef) -> str:
+        return symbol_ref
+
+    def visit_FieldAccess(self, field_access: FieldAccess, **_) -> str:
+        if str(field_access.name) in self.symbol_table:
+            dimensions = self.symbol_table[str(field_access.name)].dimensions
+        else:
+            dimensions = (True, True, True)
+
+        offset_str = self.visit(field_access.offset, dimensions=dimensions)
+
+        if field_access.data_index:
+            data_index_access = ",".join(
+                [self.visit(data_index) for data_index in field_access.data_index]
+            )
+            if offset_str == "":
+                full_string = field_access.name + f"[{data_index_access}]"
+            else:
+                full_string = field_access.name + "[" + offset_str + "," + data_index_access + "]"
+        else:
+            full_string = field_access.name + "[" + offset_str + "]"
+        return full_string
+
+    def visit_AssignStmt(self, assignment_statement: AssignStmt, **_):
+        self.body.append(
+            self.visit(assignment_statement.left) + "=" + self.visit(assignment_statement.right)
+        )
+
+    def visit_BinaryOp(self, binary: BinaryOp, **_) -> str:
+        return f"( {self.visit(binary.left)} {binary.op} {self.visit(binary.right)} )"
+
+    def visit_Literal(self, literal: Literal, **_) -> str:
+        if literal.dtype == DataType.BOOL:
+            if literal.value == BuiltInLiteral.TRUE:
+                literal_value = "True"
+            else:
+                literal_value = "False"
+        else:
+            literal_value = str(literal.value)
+
+        if literal.dtype.bit_count() != 4:
+            literal_code = f"{self.visit(literal.dtype)}({literal_value})"
+        else:
+            literal_code = literal_value
+        return literal_code
```

Comment:
early returns seem like a good fit here

```suggestion
    def visit_Literal(self, literal: Literal, **_) -> str:
        if literal.dtype == DataType.BOOL:
            return "True" if literal.value == BuiltInLiteral.TRUE else "False"

        literal_value = str(literal.value)
        if literal.dtype.bit_count() != 4:
            return f"{self.visit(literal.dtype)}({literal_value})"

        return literal_value
```

Reply:
does not quite work because we need to visit the dtype on bools also, but I pushed an adapted version